plugins/formal-verification/formver.plugin/src/org/jetbrains/kotlin/formver/ViperPoweredDeclarationChecker.kt

```diff
@@ -38,11 +53,15 @@ class ViperPoweredDeclarationChecker(private val session: FirSession, private va
 
         try {
             val verifier = Verifier()
-
-            val success = verifier.verify(program) {
-                reporter.reportOn(declaration.source, it.error, it.msg, context)
+            val onFailure = { err: VerifierError ->
+                reporter.reportOn(declaration.source, err.error, err.msg, context)
             }
 
+            val consistent = verifier.checkConsistency(program, onFailure)
+            // If the Viper program is not consistent, that's our error; we shouldn't surface it to the user as an unverified contract.
```

Comment:
I have two question about this: First even if we don't want to surface it to the user, how are we getting the consistency errors for debugging if we want to. And secondly wouldn't it be kind of confusion for the user if they just get a "contract couldn't be verified" without telling them that the problem is actually the plugin not their contract.
I suspect that there is no easy way to do this but in that case I might prefer consistency errors being reported as well for now.

Reply:
I think you're reading the comment backwards: we *do* surface these errors to the user (that's what `onFailure` does), the difference is that now we don't *also* give an error that the function may have a non-verified contract (which is confusing anyway since such a function might not even have a contract).

Reply:
Ah then this is perfect.